firrtl/src/main/scala/firrtl/ir/IR.scala

```diff
@@ -984,6 +984,15 @@ abstract class AggregateType extends Type {
   def mapWidth(f:     Width => Width): Type = this
   def foreachWidth(f: Width => Unit):  Unit = ()
 }
+case class ConstGroundType(underlying: GroundType) extends GroundType with UseSerializer{
+  val width = underlying.width
+  def mapWidth(f:     Width => Width) = underlying.mapWidth(f)
+  def foreachWidth(f: Width => Unit) = underlying.foreachWidth(f)
+}
+case class ConstAggregateType(underlying: AggregateType) extends AggregateType with UseSerializer{
+  def mapType(f: Type => Type): Type = underlying.mapType(f)
+  def foreachType(f: Type => Unit): Unit = underlying.foreachType(f)
+}
```

Comment:
I know I originally suggested having both, but can you make `ConstType` just extend `Type`? Do we need both?

Reply:
done, I don't think we need both.